assembly/src/ast/nodes/format.rs

```diff
@@ -0,0 +1,151 @@
+use super::{AstFormatterContext, FormattableCodeBody, Instruction, Node};
+use core::fmt;
+
+// FORMATTING OF NODES
+// ================================================================================================
+pub struct FormattableNode<'a> {
+    node: &'a Node,
+    context: &'a AstFormatterContext<'a>,
+}
+
+impl<'a> FormattableNode<'a> {
+    pub fn new(node: &'a Node, context: &'a AstFormatterContext<'a>) -> Self {
+        Self { node, context }
+    }
+}
+
+impl fmt::Display for FormattableNode<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self.node {
+            Node::Instruction(i) => {
+                write!(f, "{}", FormattableInstruction::new(i, self.context))
+            }
+            Node::IfElse {
+                true_case,
+                false_case,
+            } => {
+                self.context.indent(f)?;
+                writeln!(f, "if.true")?;
+
+                write!(
+                    f,
+                    "{}",
+                    FormattableCodeBody::new(true_case, &self.context.inner_scope_context())
+                )?;
+
+                self.context.indent(f)?;
+                writeln!(f, "else")?;
+
+                write!(
+                    f,
+                    "{}",
+                    FormattableCodeBody::new(false_case, &self.context.inner_scope_context())
+                )?;
```

Comment:
We could, though we should do the same for the `then` branch as well. Once the program has been parsed up we don't know whehter the original code was `if.true` or `if.false`, i.e., we don't know if the `true` branch of the `false` branch is the "main" branch.

The current code always outputs `if.true`, but if the `true` branch is empty it could be output as `if.false <false-branch> end`.